pypsa/statistics.py

```diff
@@ -127,24 +161,29 @@ def port_mask(n, c, port="", bus_carrier=None):
     if bus_carrier is None:
         mask = n.df(c)[f"bus{port}"] != ""
     elif isinstance(bus_carrier, str):
-        mask = n.df(c)[f"bus{port}"].map(n.buses.carrier).eq(bus_carrier)
+        mask = (
+            n.df(c)[f"bus{port}"]
+            .map(n.buses.carrier)
+            .str.contains(bus_carrier)
```

Comment:
could we address the "string contains" feature later?

Reply:
I already made an improvement there, but we can discuss it as soon as I am finished, and it is ready for review.